capstone-app/src/app/modules/maps/map/map.component.ts

```diff
@@ -45,6 +49,25 @@ export class MapComponent implements OnInit{
     this.locationSearch();
   }
 
+  ngOnChanges(change: SimpleChanges) {
+    if (change.activeMark) {
+      if (change.activeMark.currentValue === '') {
+        this.infoWindow.close();
+      }
+      else {
```

Comment:
Replace "else { if () {" with "else if () {". In general, see if you can reduce the number of nested ifs in this method.